packages/components/src/templates/next/components/internal/Link/utils/generateAriaLabel.ts

```diff
@@ -0,0 +1,41 @@
+import type { LinkProps } from "~/interfaces/internal/Link"
+
+type GenerateAriaLabelProps = Pick<LinkProps, "isExternal" | "label"> & {
+  textContent?: string
+}
+
+const getDomainFromUrl = (url: string) => {
+  if (!url) return ""
+
+  const urlObj = new URL(url)
+  return urlObj.hostname
+}
+
+const isUrl = (textContent: string) => {
+  if (!textContent) return false
+
+  try {
+    new URL(textContent)
+    return true
+  } catch {
+    return false
+  }
+}
+
+export const generateAriaLabel = ({
+  label,
+  textContent,
+  isExternal = false,
+}: GenerateAriaLabelProps): string | undefined => {
```

Comment:
[issue] should we pass in `href` rather than `textContent`?

Reply:
ah the textContent is the text that's being hyperlinked - it can be a non-url like [this is an example](https://www.google.com) where the textContent is `this is an example`